pkg/operators/wasm/maps.go

```diff
@@ -56,6 +68,61 @@ func (i *wasmOperatorInstance) addMapFuncs(env wazero.HostModuleBuilder) {
 		},
 		[]wapi.ValueType{wapi.ValueTypeI32}, // Error
 	)
+
+	exportFunction(env, "mapClose", i.mapClose,
+		[]wapi.ValueType{
+			wapi.ValueTypeI32, // Map
+		},
+		[]wapi.ValueType{wapi.ValueTypeI32}, // Error
+	)
+}
+
+var createdMapMutex sync.RWMutex
+var createdMap = make(map[uint32]struct{})
```

Comment:
It should be part of `wasmOperatorInstance` to avoid collisions when running several gadgets at the same time.